package/MDAnalysis/bundle.py

```diff
@@ -0,0 +1,267 @@
+                         "do not match".format(len(simulations), len(topology)))
+
+    bundle = dtr.Bundle()
+    for i, sim in enumerate(simulations):
+        bundle = add_to_bundle(bundle=bundle, simulation=sim, topology=tops[i],
+                               name=names[i])
+       
+
+    ## Add any auxiliaries
+    auxargs = aux_kwargs if aux_kwargs is not None else {}
+    if auxs is not None:
+        for auxname, auxvals in auxs.items():
+            args = aux_kwargs.get(auxname, {})
+            if len(auxvals) != len(simulations):
+                raise ValueError("Number of simulations ({}) and number of values"
+                                 " for auxiliary {} ({}) do not match".format(
+                                       len(simulations), auxname, len(auxvals)))
+            for i, sim in enumerate(bundle):
+                sim.universe.trajectory.add_auxiliary(auxname=auxname, 
+                                                      auxdata=auxvals[i], **args) 
+
+    ## Add any metadata
+    data = data if data is not None else {}
+    for key, values in data.items():
+        if (isinstance(values, (list, np.ndarray)) and 
+           len(values) != len(simulations)):
+            raise ValueError("Number of simulations ({}) and number of values for"
+                             " metadata {} ({}) do not match".format(
+                                            len(simulations), key, len(values)))
+        bundle.categories[key] = values
+
+    return bundle
+
+
+def add_to_bundle(bundle=None, simulation=None, name=None, index=None, 
+                  topology=None, univ_kwargs=None):
+    """ Insert *simulation* into *bundle* at the position *index* and return new
+    bundle.
+
+    If *bundle* is None, a new bundle is created containing just *simulation*
+    and returned. 
+
+    *simulation* may be [[ ... see above, or list here as well? ]]
+
+    Parameters
+    ----------
+    bundle : datreant Bundle object, optional
+        bundle to insert simulation into
+    simulation :
+        simulation to add; [[ list possible types? ]]
+    name : str
+        name for the simulation
+    index : int, optional
+        position in which to insert *simulation*. If *None* (default), the 
+        simulation will be added in the last position.
+    topology : str, optional
+        filename for corresponding topology if simulation is provided as a 
+        trajectory filename
+    univ_kwargs : dict, optional
+        dictionary of additional arguments to pass to Universe.
+
+
+    Returns
+    -------
+    bundle : datreant Bundle object
+        updated bundle
+    """
+    new_sim = _make_sim(simulation, name=name, topology=topology, 
+                        univ_kwargs=univ_kwargs)
+    if bundle is None:
+        # make a new bundle and return
+        # TODO - should we warn this is what we're doing?
+        return dtr.Bundle(simulation)
+    elif isinstance(bundle, dtr.Bundle):
```

Comment:
I would not test for type here. If it looks like a duck, and sounds like a duck, it's a duck.